Document/0x05i-Testing-Code-Quality-and-Build-Settings.md

```diff
@@ -553,3 +553,101 @@ class a$b
+In order to use the plugin, the following steps need to be applied:
+Install the plugin from Maven central repo by adding the following script to your build.gradle:
+
+``groovy
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'org.owasp:dependency-check-gradle:3.2.0'
+    }
+}
+
+apply plugin: 'org.owasp.dependencycheck'
+```
+
+Once gradle has invoked the plugin, you can create a report by running
+```sh
+gradle assemble
+gradle dependencyCheckAnalyze --info
+```
+The report will be in `build/reports` unless otherwise configured. Use the report in order to analyse the vulnerabilities found. See remediation on what to do given the vulnerabilities found with the libraries.
+
+Please be advised that the plugin requires to download a vulnerability feed. Consult the documentation in case issues arise with the plugin.
+
+Alternatively there are commercial tools which might have a better coverage of the dependencies found for the libraries being used, such as <#TODO: add library checkers here>. The actual result of using either the owasp dependency checker or another tool varies on the type of (NDK related or SDK related) libraries. 
+
+Lastly, please note that for hybrid applications, one will have to check the javascript dependencies. Similarly for <#TODO insert ms stuff here>, one will have to check the C# dependencies.
+
+##### Detecting the licenses used by the libraries of the application
+
+In order to ensure that the copyright laws are not infringed, one can best check the dependencies by using a plugin which can iterate over the different libraries, such as `License Gradle Plugin`. This plugin can be used by taking the following steps.
+
+In your `build.gradle` file add:
+
+```groovy
+plugins {
+    i id "com.github.hierynomus.license-report" version"{license_plugin_version"
+}
+```
+Now, after the plugin is picked up, use the following commands:
+```sh
+gradle assemble
+gradle downloadLicenses 
+```
+
+Now a license-report will be generated, which can be used to consult the licenses used by the third party libraries. Please check the license agreemts to see whether a copyright notice needs to be included into the app and whether the licensetype requires to open-source the code of the application.
+
+Similar to dependency checking, there are commercial tools which are able to check the licenses as well, such as <#TODO: add library checkers here>. 
+Note: If in doubt about the implications of a license model used by a third party library, then consult with a legal specialist. 
+
+#### Dynamic Analysis
+
+The dynamic analysis of this secion comprises validating whether the copyrights of the licensens have been adhered to. This often means that the application should have an `about` or `EULA` section in which the copy-right statements are noted as required by the license of the third party library.
+
+#### Remediation

```

Comment:
Remediation is not part of a test case anymore, there is only the dynamic and static part in a test case. 
Please put this at the end of the static analysis.